src/notation/view/internal/harppedalpopupmodel.cpp

```diff
@@ -0,0 +1,177 @@
+#include "log.h"
+
+using namespace mu::notation;
+
+HarpPedalPopupModel::HarpPedalPopupModel(QObject* parent)
+    : AbstractElementPopupModel(parent)
+{
+    setModelType(PopupModelType::TYPE_HARP_DIAGRAM);
+    setTitle("Harp pedal");
+    m_diagram = toHarpPedalDiagram(getElement());
+    m_isDiagram = m_diagram->isDiagram();
+    setPopupPedalState(m_diagram->getPedalState());
+}
+
+std::array<mu::engraving::PedalPosition, 7> HarpPedalPopupModel::getPopupPedalState()
+{
+    std::array<mu::engraving::PedalPosition, 7> posArr;
+    for (int i = 0; i < 7; i++) {
+        switch (m_pedalState.at(i)) {
+        case HarpPedalPopupModel::Position::FLAT:
+            posArr[i] = mu::engraving::PedalPosition::FLAT;
+            break;
+        case HarpPedalPopupModel::Position::NATURAL:
+            posArr[i] = mu::engraving::PedalPosition::NATURAL;
+            break;
+        case HarpPedalPopupModel::Position::SHARP:
+            posArr[i] = mu::engraving::PedalPosition::SHARP;
+            break;
+        case HarpPedalPopupModel::Position::UNSET:
+            posArr[i] = mu::engraving::PedalPosition::UNSET;
+            break;
+        default:
+            posArr[i] = mu::engraving::PedalPosition::UNSET;
+            break;
+        }
+    }
+
+    return posArr;
+}
+
+void HarpPedalPopupModel::setPopupPedalState(std::array<HarpPedalPopupModel::Position, 7> pos)
+{
+    m_pedalState = pos;
+}
+
+void HarpPedalPopupModel::setPopupPedalState(std::array<mu::engraving::PedalPosition, 7> pos)
+{
+    std::array<HarpPedalPopupModel::Position, 7> posArr;
+    for (int i = 0; i < 7; i++) {
+        switch (pos.at(i)) {
+        case mu::engraving::PedalPosition::FLAT:
+            posArr[i] = HarpPedalPopupModel::Position::FLAT;
+            break;
+        case mu::engraving::PedalPosition::NATURAL:
+            posArr[i] = HarpPedalPopupModel::Position::NATURAL;
+            break;
+        case mu::engraving::PedalPosition::SHARP:
+            posArr[i] = HarpPedalPopupModel::Position::SHARP;
+            break;
+        case mu::engraving::PedalPosition::UNSET:
+            posArr[i] = HarpPedalPopupModel::Position::UNSET;
+            break;
+        default:
+            posArr[i] = HarpPedalPopupModel::Position::UNSET;
+            break;
+        }
+    }
+
+    m_pedalState = posArr;
+}
+
+bool HarpPedalPopupModel::isDiagram() const
+{
+    return m_isDiagram;
+}
+
+QPointF HarpPedalPopupModel::pos() const
+{
+    return fromLogical(m_diagram->canvasPos()).toQPointF();
+}
+
+QPointF HarpPedalPopupModel::size() const
+{
+    RectF elemRect = fromLogical(m_diagram->canvasBoundingRect());
+    return QPointF(elemRect.width(), elemRect.height());
+}
+
+bool HarpPedalPopupModel::belowStave() const
+{
+    if (m_diagram != nullptr && (m_diagram->getProperty(mu::engraving::Pid::PLACEMENT) == mu::engraving::PlacementV::BELOW)) {
+        return true;
+    }
+    return false;
+}
+
+QPointF HarpPedalPopupModel::staffPos() const
```

Comment:
I've noticed one remaining issue - the popup tries to avoid all staves instead of just the stave the diagram is attached to as measure contains the whole system.